samples/matter/common/cmake/source_common.cmake

```diff
@@ -73,5 +73,11 @@ if(CONFIG_NCS_SAMPLE_MATTER_DIAGNOSTIC_LOGS)
 
     if(CONFIG_NCS_SAMPLE_MATTER_DIAGNOSTIC_LOGS_END_USER_LOGS)
         target_sources(app PRIVATE ${MATTER_COMMONS_SRC_DIR}/diagnostic/diagnostic_logs_end_user.cpp)
+        target_sources(app PRIVATE ${MATTER_COMMONS_SRC_DIR}/diagnostic/diagnostic_logs_retention.cpp)
+    endif()
+
+    if(CONFIG_NCS_SAMPLE_MATTER_DIAGNOSTIC_LOGS_NETWORK_LOGS)
+        target_sources(app PRIVATE ${MATTER_COMMONS_SRC_DIR}/diagnostic/diagnostic_logs_network.cpp)
+        target_sources(app PRIVATE ${MATTER_COMMONS_SRC_DIR}/diagnostic/diagnostic_logs_retention.cpp)
```

Comment:
is it a good idea to set a source twice? This will be a case when both network and user logs are enabled. I guess it can be optimized somehow here.

Reply:
Yeah, sure, I just missed it. Fixed.